### DISCUSSION TITLE
Transaction imported for last months budget gets missed by current months budget.

### DISCUSSION THREAD

**Discussion by jwmay2012** [firefly-iii/firefly-iii#8232] [Firefly III] [answered]
Been using Firefly for nearly a year and it's been awesome. Tho, I've noticing an oddity with my use case that I'm not sure how to solve.

Transactions that are made at the end of the month, but imported to firefly on the start of next month, have their withdrawals applied to last months budget (correctly), corrupting this months budget rollovers.

My bank shows pending transactions with titles that are different from the final posted transaction (with no IDs) and those "pending titles" are mostly not possible to target correctly with rules. So I have to wait until they finish posting before importing.

End result-- transaction imports are always a few days behind their purchase date, and I run my import script every few days anyways. 
So it's often the case that a new budget is generated at the start of the month, then more transactions for last month are imported/applied and the current budget doesn't know about them, becoming inflated. 

The only solution I can think of on my end is to prevent a new months budget from being generated until all of last months transactions are posted. But this means I'm budget-less and not tracking transactions for 5-7 days at the start of each month. :( 

#### Would it be reasonable for a new feature to account for this?
Such as, `when` adding a transaction to a budget-month, `if` there are newer budget-months, `then` deduct from subsequent budget-months? 

What do you think?

**Comment by JC5** [comment]
No, sorry. I'm not going to account for that situation.

**Reply by jwmay2012** [reply]
After thinking more about this, I see why not. 
Some edge cases that would have to be ignored, like modifying the transaction date later on. At best it would be a situational convenience with inconsistencies.

Maybe I can write something to adjust the budget after a webhook.

**Reply by JC5** [reply]
Webhooks would work fine! There's an API to edit those amounts, so you can post to that once a transaction matches your own logic.

**Comment by jwmay2012** [comment, marked as answer]
In case anyone else wants to remedy this--

Here's a basic Go script I slapped together that seems to work alright with creation webhooks. 
https://gist.github.com/jwmay2012/a1279cce7e80a213baaba7698cd1b39d

> If a transaction is imported this month, but is dated last month, AND has a budget, then subtract the transaction amount from this months budget limit.

Shouldn't be used with transaction update webhooks, therefore it doesn't adjust transactions that don't already have a budget assigned on creation. (aka, unknown transactions from new sources)

Does work with Budgets assigned via Rules on creation, of course.
